packages/components/src/components/segmented-control/segmented-control/segmented-control-styles.ts

```diff
@@ -1,18 +1,28 @@
-import { getCss } from '../../../utils';
+import { buildResponsiveStyles, getCss } from '../../../utils';
 import { addImportantToEachRule, hostHiddenStyles } from '../../../styles';
+import type { BreakpointCustomizable } from '../../../types';
+import { SegmentedControlColumn } from './segmented-control-utils';
 
 const MIN_ITEM_WIDTH = 46;
 const MAX_ITEM_WIDTH = 220;
 
-export const getComponentCss = (maxWidth: number): string => {
-  maxWidth = (maxWidth > MAX_ITEM_WIDTH && MAX_ITEM_WIDTH) || (maxWidth < MIN_ITEM_WIDTH && MIN_ITEM_WIDTH) || maxWidth;
+export const getComponentCss = (
+  maxItemWidth: number,
+  columns: BreakpointCustomizable<SegmentedControlColumn>
+): string => {
+  const maxWidth =
```

Comment:
Do we need new const or why not do it as it was before by sanitizing the parameter.